src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -81,7 +80,7 @@ internal class PersistentStreamPullingAgent : SystemTarget, IPersistentStreamPul
         /// <param name="queueAdapterCache"></param>
         /// <param name="failureHandler"></param>
         /// <returns></returns>
-        public async Task Initialize(Immutable<IQueueAdapter> qAdapter, Immutable<IQueueAdapterCache> queueAdapterCache, Immutable<IStreamFailureHandler> failureHandler)
+        public Task Initialize(Immutable<IQueueAdapter> qAdapter, Immutable<IQueueAdapterCache> queueAdapterCache, Immutable<IStreamFailureHandler> failureHandler)
```

Comment:
Since this method can throw exceptions that are not usage errors (according to the [TAP guidelines#Exceptions](https://msdn.microsoft.com/en-us/library/hh873175(v=vs.110).aspx#Anchor_2) ), it should probably use the `OrleansTaskExtentions.WrapInTask` utility method so that exceptions result in faulted tasks instead of failing to produce a task.